x-pack/plugins/alerting/server/rules_client/lib/mapped_params_utils.ts

```diff
@@ -0,0 +1,152 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { snakeCase } from 'lodash';
+import { AlertTypeParams, MappedParams, MappedParamsProperties } from '../../types';
+import { SavedObjectAttribute } from '../../../../../../src/core/server';
+import {
+  iterateFilterKureyNode,
+  IterateFilterKureyNodeParams,
+  IterateActionProps,
+  getFieldNameAttribute,
+} from './validate_attributes';
+
+export const MAPPED_PARAMS_PROPERTIES: Array<keyof MappedParamsProperties> = [
+  'risk_score',
+  'severity',
+];
+
+const SEVERITY_MAP: Record<string, string> = {
+  low: '20-low',
+  medium: '40-medium',
+  high: '60-high',
+  critical: '80-critical',
+};
+
+/**
+ * Returns the mapped_params object when given a params object.
+ * The function will match params present in MAPPED_PARAMS_PROPERTIES and
+ * return an empty object if nothing is matched.
+ */
+export const getMappedParams = (params: AlertTypeParams) => {
+  return Object.entries(params).reduce<MappedParams>((result, [key, value]) => {
+    const snakeCaseKey = snakeCase(key);
+
+    if (MAPPED_PARAMS_PROPERTIES.includes(snakeCaseKey as keyof MappedParamsProperties)) {
+      result[snakeCaseKey] = getModifiedValue(
+        snakeCaseKey,
+        value as string
+      ) as SavedObjectAttribute;
+    }
+
+    return result;
+  }, {});
+};
+
+/**
+ * Returns a string of the filter, but with params replaced with mapped_params.
+ * This function will check both camel and snake case to make sure we're consistent
+ * with the naming
+ *
+ * i.e.: 'alerts.attributes.params.riskScore' -> 'alerts.attributes.mapped_params.risk_score'
+ */
+export const getModifiedFilter = (filter: string) => {
+  return filter.replace('.params.', '.mapped_params.');
+};
+
+/**
+ * Returns modified field with mapped_params instead of params.
+ *
+ * i.e.: 'params.riskScore' -> 'mapped_params.risk_score'
+ */
+export const getModifiedField = (field: string | undefined) => {
+  if (!field) {
+    return field;
+  }
+
+  const sortFieldToReplace = `${snakeCase(field.replace('params.', ''))}`;
+
+  if (MAPPED_PARAMS_PROPERTIES.includes(sortFieldToReplace as keyof MappedParamsProperties)) {
+    return `mapped_params.${sortFieldToReplace}`;
+  }
+
+  return field;
+};
+
+/**
+ * Returns modified search fields with mapped_params instead of params.
+ *
+ * i.e.:
+ * [
+ *    'params.riskScore',
+ *    'params.severity',
+ * ]
+ * ->
+ * [
+ *    'mapped_params.riskScore',
+ *    'mapped_params.severity',
+ * ]
+ */
+export const getModifiedSearchFields = (searchFields: string[] | undefined) => {
+  if (!searchFields) {
+    return searchFields;
+  }
+
+  return searchFields.reduce<string[]>((result, field) => {
+    const modifiedField = getModifiedField(field);
+    if (modifiedField) {
+      return [...result, modifiedField];
+    }
+    return result;
+  }, []);
+};
+
+export const getModifiedValue = (key: string, value: string) => {
```

Comment:
Ah okay so to avoid needing to store the label and the value, we're going to put the value in the label so we can do the right sort. We should add a clarifying comment for future folks though